### PR TITLE
Added new module l10n_nl_cbs_export

### PR DESCRIPTION
**Pull request by ploegvde**
@astirpe  As discussed in issue #23. This is they CBS export module we use at one of our customers. The module creates a cron job and creates the export file automatically every month. We have added some users as followers of the CBS export so that they get notified.

The output file has to be manually uploaded on the CBS site. There is no other way to my knowledge.

In the module some values of fields are hard codes and need configuration, so that the module is suitable for other companies. I have no problem in doing that. 

So the question is, if the module is something we can use as a start?

### CODE REVIEW

**Comment by astirpe** [comment]
@ploegvde thank you for this, indeed is a good start!
Before starting the review, could you work on it, in order to make the module more generic and configurable?
Let me know in case you need help.

**Comment by ploegvde** [comment]
yes, i will do that. No problem.

File: l10n_nl_cbs_export/LICENSE
Comment (hbrunn) [inline l10n_nl_cbs_export/LICENSE]: the licence key is enough imho

File: l10n_nl_cbs_export/__manifest__.py
Comment (hbrunn) [inline l10n_nl_cbs_export/__manifest__.py]: this is inconsistent with the license file, do you mean agpl-3 here?

**Comment by ploegvde** [comment]
@astirpe Hi, i checked the module and the output. In my branch

https://github.com/Odoo-experts/l10n-netherlands/tree/l10n_nl_cbs_export

I have added documentation of CBS. It is only in Dutch.

What i see is that the fixed values we use are for most common situations correct. There is only 1 field that needs to be added to product template and that is "Bijzondere maatstaf"

But what i'm doubting about is the use of the "intrastat-code". In Odoo V10 we have now also the HS-Code and to me it is not clear what the difference is. So maybe others can help on this.

So think the module can get a fitst review based on what it does now?

File: l10n_nl_cbs_export/__manifest__.py
Comment (hbrunn) [inline l10n_nl_cbs_export/__manifest__.py]: remove empty keys and ones with the defaul value

File: l10n_nl_cbs_export/__manifest__.py
Comment (hbrunn) [inline l10n_nl_cbs_export/__manifest__.py]: remove this and use a [README.rst](https://github.com/OCA/maintainer-tools/blob/master/template/module/README.rst)

File: l10n_nl_cbs_export/data/cron_process_cbs_export.xml
Comment (hbrunn) [inline l10n_nl_cbs_export/data/cron_process_cbs_export.xml]: you don't need any of those evals, replace them by the plain value

File: l10n_nl_cbs_export/models/__init__.py
Comment (hbrunn) [inline l10n_nl_cbs_export/models/__init__.py]: use relative imports

File: l10n_nl_cbs_export/__init__.py
Comment (hbrunn) [inline l10n_nl_cbs_export/__init__.py]: use relative imports

File: l10n_nl_cbs_export/__manifest__.py
Comment (hbrunn) [inline l10n_nl_cbs_export/__manifest__.py]: add OCA as first author

File: l10n_nl_cbs_export/models/cbs_export_file.py
Comment (hbrunn) [inline l10n_nl_cbs_export/models/cbs_export_file.py]: this is not a list of ids, but a recordset

File: l10n_nl_cbs_export/models/cbs_export_file.py
Comment (hbrunn) [inline l10n_nl_cbs_export/models/cbs_export_file.py]: `account_ids` is a very misleading name for a variable holding a recordset of invoices

File: l10n_nl_cbs_export/models/cbs_export_file.py
Comment (hbrunn) [inline l10n_nl_cbs_export/models/cbs_export_file.py]: replace all of this by a qweb template that does all the formatting etc (compare with the xaf export)

**Review by hbrunn — CHANGES_REQUESTED** [review]
There are also a lot of complaints from runbot: https://travis-ci.org/OCA/l10n-netherlands/jobs/240298118#L556-L580

I think you can save yourself some inconvenience by using a date field or (if the reporting period can be something else than months) date ranges. Then you save those files and jump through some hoops to make this unique (shouldn't there be a uniqueness constraint for good measure?), is there a reason to do this or can't you just generate those on the fly as requested?

**Comment by astirpe** [comment]
@ploegvde great, thanks! I will start a review as soon as I have some time.

There is also an OCA module [product_harmonized_system](https://github.com/OCA/intrastat/tree/10.0/product_harmonized_system) that handles the HS codes.

File: l10n_nl_cbs_export/view/cbs_export_file_view.xml
Comment (astirpe) [inline l10n_nl_cbs_export/view/cbs_export_file_view.xml]: Instead of "account Invoice", this tab could be named in a more user friendly way, eg.: "Invoices".

File: l10n_nl_cbs_export/view/cbs_export_file_view.xml
Comment (astirpe) [inline l10n_nl_cbs_export/view/cbs_export_file_view.xml]: I manually created a record of model `cbs.export.file` without any export file. The button "CBS Export Again" remains visible, I was expecting it to be hidden because of this:
`
attrs="{'invisible': [('cbs_export_invoice', '!=', False)]}"
`
so it seems to me that the invisible attribute is not working.

However I think you can remove the invisible attribute at all, so that the button will be always displayed. In this case just rename the button "CBS Export Again" to something like "Manual CBS Export".

You can also display the button only for a certain group (eg.: Account Advisor) by adding a `groups=" ... "` definition.

File: l10n_nl_cbs_export/models/cbs_export_file.py
Comment (ploegvde) [reply l10n_nl_cbs_export/models/cbs_export_file.py]: Why would this be needed?

Comment (hbrunn) [reply l10n_nl_cbs_export/models/cbs_export_file.py]: it gives other developers a modular way to hook into the generation process and modify it. So it's needed to make your code better.

Comment (ploegvde) [reply l10n_nl_cbs_export/models/cbs_export_file.py]: I would like the module first to be ready and functional. If it needs improvement, then we improve later. I like users to benefit of what now is available. Is that possible?

Comment (astirpe) [reply l10n_nl_cbs_export/models/cbs_export_file.py]: Considering that this is the first @ploegvde 's contribution to the OCA repository (if I'm not wrong) and in the spirit of the OCA principles, I think that the proposal of Erwin can be accepted. I will convert this formatting part to a qweb template later on, after this PR is merged, in case nobody wants to do it.

File: l10n_nl_cbs_export/security/ir.model.access.csv
Comment (astirpe) [inline l10n_nl_cbs_export/security/ir.model.access.csv:1]: I don't see any group with creation rights for model cbs.export.file.
At least group "Account Advisor" must have creation/write rights.

**Review by astirpe — CHANGES_REQUESTED** [review]

**Review by hbrunn — APPROVED** [review]

**Review by astirpe — APPROVED** [review]

**Comment by ploegvde** [comment]
Can this be merged?

**Pull request merged**
